src/Report/ReportInterface.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Report;
+
+use Symfony\Component\Stopwatch\Stopwatch;
+
+interface ReportInterface
+{
+    /**
+     * @param int $showAppliedFixers
+     */
+    public function setShowAppliedFixers($showAppliedFixers);
+
+    /**
+     * @param bool $needDiff
+     */
+    public function setShowDiff($needDiff);
+
+    /**
+     * @param bool $dryRun
+     */
+    public function setDryRun($dryRun);
+
+    /**
+     * @param bool $decoratedOutput
+     */
+    public function setDecoratedOutput($decoratedOutput);
+
+    /**
+     * Sets the Stopwatch.
+     *
+     * @param Stopwatch $stopwatch
+     */
+    public function setStopwatch(Stopwatch $stopwatch);
```

Comment:
You just won't have 
```
$reportClass = self::$reports[$format];
new $reportClass();
```

anymore, but full factory class that will handle creation of reports.